docs/knowledge-base.rst

```diff
@@ -0,0 +1,107 @@
+
+    - find all aliases of gene ``BRCA2``
+    - finding all genes of type ``protein_coding``
+    - find all transcripts of gene ``FHIT``
+    - converting ``gene_id`` to ``gene_symbol``
+    - ...
+
+
+Feature
+=======
+``Feature`` object represents a genomic feature: a gene, a transcript, an
+exon, etc. You can search and filter ``Feature`` objects by ``feature``
+endpoint, similarly like ``Data``, ``Sample`` or any other ReSDK resource::
+
+    feature1 = res.feature.get(feature_id="BRCA2")
+
+To examine all attributes of a ``Feature``, see the :ref:`reference`.
+Here we will list a few::
+
+    # Get the feature:
+    feature1 = res.feature.get(feature_id="BRCA2")
+
+    # Database where this feature is defined, e.g. ENSEMBL, UCSC, NCBI, ...
+    feature1.source
+
+    # Unique identifier of a feature
+    feature1.feature_id
+
+    # Feature species
+    feature1.species
+
+    # Feature type, e.g. gene, transcript, exon, ...
+    feature1.type
+
+    # Feature name
+    feature1.name
+
+    # List of feature aliases
+    feature1.aliases
+
+But the real power is in the search / filter capabilities. Let's just
+list some examples::
+
+    # Count number of Human "protein-conding" transcripts in ENSEMBL database
+    res.feature.filter(
+        species="Homo sapiens",
+        type="transcript",
+        subtype="protein-coding",
+        source="ENSEMBL",
+    ).count()
+
+    # Convert all gene_id's in a list ``gene_ids`` to gene symbols::
+    genes = res.feature.filter(
+        feature_id__in=gene_ids,
+        type="gene",
+        species="Homo sapiens",
+    )
+    gene_symbols = [g.name for g in genes]
+
+
+Mapping
+=======
+Mapping is a *connection* between two features. Features can be related
+in various ways - we have different types of relations between features.
+This type is indicated by ``relation_type`` attribute. It can be one of
+the following options:
+
+    - Type: ``crossdb``. Two features from different sources (databases)
+      that describe same feature. Example: connection for gene BRCA2
+      between database "UniProtKB" and "UCSC"
+    - Type: ``ortholog``. Two features from different species that
+      describe orthologous gene
+    - Type: ``transcript``. Connection between gene and it's transcripts
+    - Type: ``exon``. Connection between gene / transcript and it's exons
+
+Again, we will only list some examples and than let you imagination fly::
+
+    # Find UniProtKB ID for gene with given ENSEMBL ID:
+    mapping = res.mapping.get(
+        relation_type="crossdb",
+        source_id="ENSG00000189283",
+        source_db="ENSEMBL",
+        target_db="UniProtKB",
+        source_species="Homo sapiens",
+        target_species="Homo sapiens",
+    )
+    uniprot_id = mapping.target_id
+
+    # Find mouse ortholog for Human BRCA2 gene:
+    mapping = res.mapping.get(
+        relation_type="ortholog",
+        source_id="ENSG00000139618",
```

Comment:
I know I am not a biologist, but `BRCA2` and `ENSG00000139618` don't look to similar.